### PR TITLE
[#163084103] Exclude rdsadmin queries from max_tx_age metric, create new max_system_tx_age metric

### PR DESCRIPTION
**Pull request by AP-Hunt**
# What
Excludes queries run by non-system-administrative users from the data used to create the max_tx_age metric. Introduces a new max_system_tx_age which includes only queries run by system-administrative users.

# How to review
Code review
Run it down the pipeline using the paas-cf [branch](https://github.com/alphagov/paas-cf/pull/1773) and check that the max_tx_age metric is still being published, and that the new max_system_tx_age metric is available.
    1. `cf install-plugin log-cache`
    2. `cf target -o admin -s billing`
    3. `cf tail $(cf service billing-db --guid) -f`

Paired with @tnwhitwell and verified that the queries work and produce the intended results. The tests also assert that. Probably no need for the reviewer to check again.

# Who can review
Not @AP-Hunt or @tnwhitwell

### CODE REVIEW

**Comment by AP-Hunt** [comment]
Integration tests will probably keep failing until [paas-cf#1745](https://github.com/alphagov/paas-cf/pull/1745) is merged and deployed, because they rely on it.

**Comment by AP-Hunt** [comment]
Blocked by https://github.com/alphagov/paas-rds-broker/pull/97

File: pkg/collector/postgres_collector.go [hunk 1]
```diff
@@ -140,9 +140,13 @@ var postgresMetricQueries = []metricQuery{
...
 			SELECT
-				EXTRACT(epoch FROM MAX(now() - xact_start))::INT as max_tx_age
-			FROM pg_stat_activity
-      WHERE state IN ('idle in transaction', 'active')
+				COALESCE(EXTRACT(epoch FROM MAX(now() - a.xact_start))::INT, 0) as max_tx_age
+			FROM pg_stat_activity a
+			INNER JOIN pg_roles r ON r.rolname = a.usename
+			INNER JOIN pg_group g ON r.oid = ANY (g.grolist)
+			WHERE
+				g.groname LIKE '%_manager'
```
Comment (bandesz) [inline pkg/collector/postgres_collector.go]: be more explicit here and say `'LIKE 'rdsbroker_%_manager''`

File: pkg/collector/postgres_collector.go [hunk 2]
```diff
@@ -151,6 +155,24 @@ var postgresMetricQueries = []metricQuery{
...
 	},
+	&columnMetricQuery{
+		Query: `
+			SELECT
+				COALESCE(EXTRACT(epoch FROM MAX(now() - xact_start))::INT, 0) as max_system_tx_age
+			FROM pg_stat_activity a
+			INNER JOIN pg_roles r ON r.rolname = a.usename
+			LEFT JOIN pg_group g ON r.oid = ANY (g.grolist)
+			WHERE
+				(g.groname NOT LIKE '%_manager' OR g.groname IS NULL)
```
Comment (bandesz) [inline pkg/collector/postgres_collector.go]: be more explicit here and say `NOT LIKE 'rdsbroker_%_manager''`

File: pkg/collector/postgres_collector_test.go [hunk 3]
```diff
@@ -76,6 +77,13 @@ var _ = Describe("NewPostgresMetricsCollectorDriver", func() {
 				('Code Name: K.O.Z.', '2015-02-13', 'crime', 114)
 		`)
 		Expect(err).NotTo(HaveOccurred())
+
+		groupName = fmt.Sprintf("%s_manager", utils.RandomString(10))
```
Comment (bandesz) [inline pkg/collector/postgres_collector_test.go]: What is this group used for? The purpose is not clear for me.

**Pull request merged**